src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -307,28 +308,45 @@ class SlickDataAccess(databaseConfig: Config, val dataAccess: DataAccessComponen
     runTransaction(action)
   }
 
-  override def getWorkflowsByState(states: Traversable[WorkflowState]): Future[Traversable[WorkflowInfo]] = {
+  override def getWorkflow(workflowId: WorkflowId): Future[WorkflowDescriptor] = {
+    val action = for {
+      workflowExecutionResult <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
```

Comment:
Yes it will throw and that will result in a failed `Future`.

```
scala> Seq().head
java.util.NoSuchElementException: head of empty list
  at scala.collection.immutable.Nil$.head(List.scala:420)
  at scala.collection.immutable.Nil$.head(List.scala:417)
  ... 33 elided
```

This particular pattern is in many places throughout `SlickDataAccess`, IMHO one blanket explanation would for these would be preferable to one per each instance.